reports/gitlab-commit-protocol/gitlab-commit-protocol.tex

```diff
@@ -379,13 +379,27 @@ \section{Gitaly problems}
 in \textit{storage repositories} table.
 \item  \emph{\href{https://gitlab.com/gitlab-org/gitaly/-/merge_requests/3494}{ MR about considering the primary node modified}}. 
 The ideas in this MR can be studied to find inconsistency scenarios that changes cannot solve.
-\item  \emph{\href{//gitlab.com/gitlab-org/gitaly/-/blob/master/doc/virtual\_storage.md\#known-problems}{Gitaly team known problems}} 
+\item  \emph{\href{https://gitlab.com/gitlab-org/gitaly/-/blob/master/doc/virtual\_storage.md\#known-problems}{Gitaly team known problems}} 
 that connected with sql primary election strategy. 
 \end{itemize}
 
+After Gitaly commit protocol analysis, was noticed that the incrementing of generation in \textit{repositories} 
+and \textit{store\_repositories} tables happens after Praefect (TM) 
+response to Gitaly (RM) on \textit{VoteTransactionRequest}. If the TM crashes 
+after sending \textit{Commit} response to Gitaly nodes and before transaction 
+finalizer that calls incrementing mentioned above we can hit the situation:
+	
+\begin{enumerate}
+\item Transaction was successfully finished because of quorum was reached.
```

Comment:
This sentence seems to be incorrect: if Praefect fails, the transaction can not be finished successfully. Actually, we have this situation: the inconsistency could happen in this situation 1) if at least a quorum of Gitaly nodes prepared and voted for the transaction to commit and at least one node aborted the transaction; 2) Praefect respond to Gitaly vote request with commit message; 3) Praefect crashes after responding on this request and can't update generation tables.